### PR TITLE
Feature sorting based on experiment value

### PR DESCRIPTION
**Pull request by upendras-ltts**
To Create an experiment called item_sorting
a. Add 3 groups: control, test oldest, test newest.
b. Add a single parameter sort_order, which is 
“default”, “newest_first” or “oldest_first”
c. Implementing the different sorting by checking 
the experiment and parameter value for the user

Minor code cleaning and lift the state for usefeature.

TODO's with oldest first

experiment_oldest_first

TODO's with newest first
experiment_newest_first

### CODE REVIEW

File: React-Todo-list-main/src/App.js [hunk 1]
```diff
@@ -2,35 +2,41 @@ import "./App.css";
 import { StatsigProvider } from "statsig-react";
 import { useState } from "react";
 import { TodoWrapper } from "./Components/TodoWrapper";
-
-const SS_CLIENT_SDK_KEY = "client-wWQtJlfJfEnLSIuP6QcnTATHyGVHUnJ2URTGLOVrwrl";
+import { SS_CLIENT_SDK_KEY } from "./Constant";
```
Comment (daniel-statsig) [inline React-Todo-list-main/src/App.js:5]: Please avoid shorting Statsig to SS. CLIENT_SDK_KEY is sufficient here.

File: React-Todo-list-main/src/App.js [hunk 2]
```diff
@@ -2,35 +2,41 @@ import "./App.css";
...
 
 /** Pass your StatsigUser into the StatsigProviderreturn (
```
Comment (daniel-statsig) [inline React-Todo-list-main/src/App.js:7]: Pass your StatsigUser into the StatsigProviderreturn -> Pass your StatsigUser into the StatsigProvider

File: React-Todo-list-main/src/App.js [hunk 3]
```diff
@@ -2,35 +2,41 @@ import "./App.css";
...
-  >
-    <div className="App">
-      <TodoWrapper />
-    </div>
+      sdkKey={SS_CLIENT_SDK_KEY}
+      waitForInitialization={true}
+      options={{
+        environment: { tier: "staging" },
+      }}
+      user={user}
```
Comment (daniel-statsig) [inline React-Todo-list-main/src/App.js:35]: ❤️ Thank you for the formatting

File: React-Todo-list-main/src/Components/TodoWrapper.js [hunk 4]
```diff
@@ -23,12 +63,36 @@ export const TodoWrapper = () => {
...
+    let isNewestFirst = sortingOrder === NEWEST_FIRST;
+    const sortedTodos = [...todos].sort((a, b) => {
+      if (a.createdDate < b.createdDate) return isNewestFirst ? 1 : -1;
+      if (a.createdDate > b.createdDate) return isNewestFirst ? -1 : 1;
+      return 0;
+    });
+
+    console.log(`Sorted todos : ${JSON.stringify(sortedTodos)}`);
+    return sortedTodos;
+    //setTodos(sortedTodos);
```
Comment (daniel-statsig) [inline React-Todo-list-main/src/Components/TodoWrapper.js:93]: nit: remove

File: React-Todo-list-main/src/Components/TodoWrapper.js [hunk 5]
```diff
@@ -113,10 +200,16 @@ export const TodoWrapper = () => {
 
   return (
     <div className="TodoWrapper">
-      <h1>Get Things Done !</h1>
+      <div className="header" style={{ display: "flex" }}>
+        <img
+          src="https://statsig.com/images/horz_logo.svg"
+          style={{ height: "40px", marginRight: "20px" }}
+        ></img>
+        <h1>TODO's</h1>
```
Comment (daniel-statsig) [inline React-Todo-list-main/src/Components/TodoWrapper.js:208]: TODO's -> TODOs

Comment (upendras-ltts) [reply React-Todo-list-main/src/Components/TodoWrapper.js:208]: Yes will do this

**Comment by upendras-ltts** [comment]
Thanks for review.

**Pull request merged**